lib/GameConstants.cpp

```diff
@@ -160,3 +162,30 @@ std::string ETerrainType::toString() const
 	ss << *this;
 	return ss.str();
 }
+
+std::ostream & operator<<(std::ostream & os, const EPathfindingLayer actionType)
+{
+	static const std::map<EPathfindingLayer::EEPathfindingLayer, std::string> pathfinderLayerToString =
+	{
+	#define DEFINE_ELEMENT(element) {EPathfindingLayer::element, #element}
+		DEFINE_ELEMENT(WRONG),
+		DEFINE_ELEMENT(AUTO),
+		DEFINE_ELEMENT(LAND),
+		DEFINE_ELEMENT(SAIL),
+		DEFINE_ELEMENT(WATER),
+		DEFINE_ELEMENT(AIR),
+		DEFINE_ELEMENT(NUM_LAYERS)
+
+	};
+
+	auto it = pathfinderLayerToString.find(actionType.num);
+	if (it == pathfinderLayerToString.end()) return os << "<Unknown type>";
+	else return os << it->second;
+}
+
+std::string EPathfindingLayer::toString() const
+{
+	std::stringstream ss;
+	ss << *this;
+	return ss.str();
+}

```

Comment:
I don't quite get why do you ever want to use string map here? If all you do is to write logs, case would be sufficient (and faster).